From 84c5279547e3a6f82984aa4b265b9e3816441682 Mon Sep 17 00:00:00 2001
From: Sudharsan Eswaran <eswaran@qti.qualcomm.com>
Date: Tue, 19 May 2026 10:52:55 +0530
Subject: [PATCH] Handle EP_MOVE channel change reason code

Channel change event received due to backhaul STA move is handled
properly and event is processed to store new channel in DB

Signed-off-by: Sudharsan Eswaran <eswaran@qti.qualcomm.com>
---
 common/beerocks/bwl/whm/ap_wlan_hal_whm.cpp | 27 +++++++++++++++------
 1 file changed, 19 insertions(+), 8 deletions(-)

diff --git a/common/beerocks/bwl/whm/ap_wlan_hal_whm.cpp b/common/beerocks/bwl/whm/ap_wlan_hal_whm.cpp
index a67bb1953..60c1d4207 100644
--- a/common/beerocks/bwl/whm/ap_wlan_hal_whm.cpp
+++ b/common/beerocks/bwl/whm/ap_wlan_hal_whm.cpp
@@ -1467,6 +1467,7 @@ bool ap_wlan_hal_whm::process_radio_channel_change_event(const AmbiorixVariant *
 {
 
     auto parameters = value->find_child("Updates");
+
     if (!parameters || parameters->empty()) {
         LOG(ERROR) << "Received event without Updates parameter";
         return false;
@@ -1476,20 +1477,30 @@ bool ap_wlan_hal_whm::process_radio_channel_change_event(const AmbiorixVariant *
         LOG(ERROR) << "Received event without ChannelChangeReason parameter" << chan_change_reason;
         return false;
     }
-    if (chan_change_reason != "MANUAL" && chan_change_reason != "AUTO") {
-        LOG(ERROR) << "chan_change_reason other than MANUAL or AUTO:" << chan_change_reason;
 
-        if (chan_change_reason != "DFS" || !m_accept_dfs_channel_change_after_cac_failure) {
+    if (chan_change_reason != "MANUAL" && chan_change_reason != "AUTO") {
+        LOG(INFO) << "chan_change_reason other than MANUAL or AUTO:" << chan_change_reason;
+
+        // Handle DFS channel changes
+        if (chan_change_reason == "DFS" && m_accept_dfs_channel_change_after_cac_failure) {
+            LOG(INFO) << "Channel change event received following a failed CAC_Completed, "
+                      << "handling it as part of the radar flow...";
+            m_accept_dfs_channel_change_after_cac_failure = false;
+            m_radio_info.last_csa_sw_reason               = ChanSwReason::Radar;
+        }
+        // Handle EP_MOVE (backhaul STA roaming to different channel/AP)
+        else if (chan_change_reason == "EP_MOVE") {
+            LOG(INFO) << "Channel change event received due to backhaul STA endpoint move";
+            m_radio_info.last_csa_sw_reason = ChanSwReason::Unknown;
+        }
+        else {
+            LOG(ERROR) << "Unsupported channel change reason: " << chan_change_reason;
             return false;
         }
-
-        LOG(INFO) << "Channel change event received following a failed CAC_Completed, "
-                  << "handling it as part of the radar flow...";
-        m_accept_dfs_channel_change_after_cac_failure = false;
-        m_radio_info.last_csa_sw_reason               = ChanSwReason::Radar;
     } else {
         m_radio_info.last_csa_sw_reason = ChanSwReason::Unknown;
     }
+
     event_queue_push(Event::CSA_Finished);
     return true;
 }
-- 
2.34.1

